### PR TITLE
Read a local feedback  js file

### PR DESCRIPTION
**Pull request by aabughosh**
1 need to download the feedback file
2 need to convert it to js by running the command ./tnf generate catalog feedbackjs -f path-to-feedbackjson/feedback.json -o outputpath

### CODE REVIEW

File: cnf-certification-test/results/html/feedback.js [hunk 1]
```diff
@@ -0,0 +1 @@
...ge-platform-alteration-ocp-lifecycle":"","source-all-platform-alteration-ocp-node-os-lifecycle":"","source-telco-platform-alteration-ocp-node-os-lifecycle":"","source-nontelco-platform-alteration-ocp-node-os-lifecycle":"","source-extended-platform-alteration-ocp-node-os-lifecycle":"","source-faredge-platform-alteration-ocp-node-os-lifecycle":"","source-all-platform-alteration-service-mesh-usage":"","source-telco-platform-alteration-service-mesh-usage":"","source-nontelco-platform-alteration-service-mesh-usage":"","source-extended-platform-alteration-service-mesh-usage":"","source-faredge-platform-alteration-service-mesh-usage":"","source-all-platform-alteration-sysctl-config":"","source-telco-platform-alteration-sysctl-config":"","source-nontelco-platform-alteration-sysctl-config":"","source-extended-platform-alteration-sysctl-config":"","source-faredge-platform-alteration-sysctl-config":"","source-all-platform-alteration-tainted-node-kernel":"","source-telco-platform-alteration-tainted-node-kernel":"","source-nontelco-platform-alteration-tainted-node-kernel":"","source-extended-platform-alteration-tainted-node-kernel":"","source-faredge-platform-alteration-tainted-node-kernel":""}
```
Comment (rdavid) [inline cnf-certification-test/results/html/feedback.js:1]: A new line at the end in needed.

**Review by sebrandon1 — CHANGES_REQUESTED** [review]
Needs a README update with how to use.

File: README.md [hunk 2]
```diff
@@ -63,6 +63,19 @@ you we created a parser that is local. You can open it by copying the path /
 browser then upload the claim.json file of that run.
 ![overview](docs/assets/images/htmlpage.png)
 
+## Upload a previous feedback to the html parser page
+
+A new feature that we did that allow users to upload a previous feedback that they
```
Comment (ramperher) [inline README.md:68]: Suggestion: a reader that is not aware of the project may find difficulties with trying to guess what's the "new feature you did", probably here it's better to add some links to the feature (the HTML report) and put the correct names.

**Comment by ramperher** [comment]
> 1 need to download the feedback file 2 need to convert it to js by adding feedback= at the beginning of the file 3 need to copy it to the right place

I fear this process is really really manual, is there any way of providing a script in your side that already does it, then the users only have to run that script?

File: README.md (hunk 2 above)
Comment (aabughosh) [reply README.md:68]: we have explanation about the html parser so i wrote that comments right after them

**Comment by aabughosh** [comment]
> 

i did that please see comments

File: README.md [hunk 3]
```diff
@@ -63,6 +63,18 @@ you we created a parser that is local. You can open it by copying the path /
...
+
+A feature on the html parser page that we did to allow users to upload a previous
+feedback that they have to the parser page, they have an option to download the
+feedback then to upload it for the next time from the html page another option users
+can have that feedback file in the path #your-code-path/cnf-certification-test/script/
+and it need to be named as feedback.js users need to convert the downloaded
+feedback.json to feedback.js by running the command
+./tnf generate catalog feedbackjs -f path-to-feedbackjson/feedback.json
+the feedback.js will be copy to #your-code-path/cnf-certification-test/script/
+users can see there feedback without upload it from the html page.
```
Comment (sebrandon1) [inline README.md:76]:
```suggestion
A feature on the HTML parser page that we did to allow users to upload a previous
feedback that they have to the parser page, they have an option to download the
feedback then to upload it for the next time from the HTML page another option users
can have that feedback file in the path #your-code-path/cnf-certification-test/script/
and it need to be named as `feedback.js` users need to convert the downloaded
`feedback.json` to `feedback.js` by running the command:
`./tnf generate catalog feedbackjs -f path-to-feedbackjson/feedback.json`

The `feedback.js` will be copy to #your-code-path/cnf-certification-test/script/
users can see there feedback without upload it from the HTML page.
```

File: cnf-certification-test/results/html/feedback.js [hunk 4]
```diff
@@ -0,0 +1,422 @@
+feedback={
+  "source-all-access-control-cluster-role-bindings": "dadasda",
+  "source-all-access-control-container-host-port": "dadsasd",
```
Comment (sebrandon1) [inline cnf-certification-test/results/html/feedback.js:3]:
```suggestion
  "source-all-access-control-cluster-role-bindings": "",
  "source-all-access-control-container-host-port": "",
```

**Review by sebrandon1 — COMMENTED** [review]
Just some minor comments.  Looks good!  I like the generate feature.

**Review by sebrandon1 — APPROVED** [review]

File: cmd/tnf/generate/catalog/catalog.go [hunk 5]
```diff
@@ -291,8 +299,54 @@ func runGenerateMarkdownCmd(_ *cobra.Command, _ []string) error {
...
+func runGeneratefeedbackjs(_ *cobra.Command, _ []string) error {
+	feedbackjsonFileTextPtr := &feedbackjson
+	dat, err := os.ReadFile(*feedbackjsonFileTextPtr)
+	if err != nil {
+		log.Fatalf("Error reading claim file :%v", err)
+	}
+	var obj map[string]interface{}
+	err = json.Unmarshal(dat, &obj)
+	if err != nil {
+		fmt.Println("Error:", err)
```
Comment (greyerof) [inline cmd/tnf/generate/catalog/catalog.go:328]: There's no need to use these `fmt.Println()` before returning the error, as the error is always printed. Just create a helpful fmt.Errorf("...") message.

File: cmd/tnf/generate/catalog/catalog.go [hunk 6]
```diff
@@ -291,8 +299,54 @@ func runGenerateMarkdownCmd(_ *cobra.Command, _ []string) error {
 
 // Execute executes the "catalog" CLI.
 func NewCommand() *cobra.Command {
+	generatefeedbackjs.Flags().StringVarP(
+		&feedbackjson, "feedback", "f", "",
+		"feedback.json file to convert it to js file, output is js ",
```
Comment (greyerof) [inline cmd/tnf/generate/catalog/catalog.go:304]:
```suggestion
		"path to the feedback.json file",
```

File: cmd/tnf/generate/catalog/catalog.go [hunk 7]
```diff
@@ -32,6 +33,13 @@ import (
 )
 
 var (
+	feedbackjson string
+	// generateCmd is the root of the "catalog generate" CLI program.
+	generatefeedbackjs = &cobra.Command{
+		Use:   "feedbackjs",
+		Short: "Generates feedbackjs from feedbackjson file and copy it into /cnf-certification-test/results/html",
```
Comment (greyerof) [inline cmd/tnf/generate/catalog/catalog.go:40]:
```suggestion
		Short: "Generates a javascript file called feedback.js from a feedback.json that was downloaded from the results html viewer",
```
By the way, the flags "Long" and "Example" flags can also be used to include more detailed info about both files and some examples.

File: cnf-certification-test/results/html/results.html [hunk 8]
```diff
@@ -777,7 +778,7 @@
           fillResults(json.claim.results, classification, '#mandatory-telco-table', '#optional-telco-table', "telco")
           fillResults(json.claim.results, classification, '#mandatory-non-telco-table', '#optional-non-telco-table', "nontelco")
           fillResults(json.claim.results, classification, '#mandatory-extended-table', '#optional-extended-table', "extended")
-
+          fillFeedback(feedback)
```
Comment (greyerof) [inline cnf-certification-test/results/html/results.html:781]: Shouldn't the `cnf-certification-test/results/html/results-embed.html` file be updated as well?

**Review by greyerof — COMMENTED** [review]
Some more comments:
1. I don't think this new tool should be a subcommand for the "generate catalog". I believe it's more apropriate to create a new "generate feedback-js". This involves moving the command and its functions to a new folder `cmd/tnf/generate/feedback_js`.
2. There's no need to create the `feedback.js` file inside the cnf-certification-test/results/html folder, as this is a temporary file with user input data that won't be neither uploaded to the repo nor included in the results tar.gz file. 
3. I would create another optional flag "-o" just to let the user to choose the output folder where the feedback.js file wil be created. If it's not provided, the current working directory should be ok.
```
./tnf generate feedback-js -f path/to/downloaded/feedback.json -o path/to/desired/output/folder
```
4. It seems to me that all the parser/viewer html related stuff is growing a lot lately, with an increasing number of files and steps. Maybe it is worth adding a new section in the docs folder explaining all these steps to review results, create the feedback & visualize the feedback. Could be done in further PR, anyways.

File: README.md [hunk 9]
```diff
@@ -63,6 +63,19 @@ you we created a parser that is local. You can open it by copying the path /
 browser then upload the claim.json file of that run.
 ![overview](docs/assets/images/htmlpage.png)
 
+## Upload a previous feedback to the html parser page
+
+A feature on the HTML parser page that we did to allow users to upload a previous
```
Comment (ramperher) [inline README.md:68]: I think these two new paragraphs need to be reviewed and rephrased. My feeling is that this is a bit hard to read and understand right now, honestly.

File: cnf-certification-test/results/html/results.html (hunk 8 above)
Comment (aabughosh) [reply cnf-certification-test/results/html/results.html:781]: yah we have it also as upload its ok, we have defult value that nothing in all

File: cnf-certification-test/results/html/feedback.js
[the remark this answers is not in the document]
Comment (aabughosh) [reply cnf-certification-test/results/html/feedback.js:1]: i think its better to be here as a default  and every filed on it has nothing

File: README.md [hunk 10]
```diff
@@ -63,6 +63,20 @@ you we created a parser that is local. You can open it by copying the path /
...
+feedback that they have to the parser page, they have an option to download the
+feedback then to upload it for the next time from the HTML page another option users
+can have that feedback file in the path #your-code-path/cnf-certification-test/script/
+and it need to be named as `feedback.js` users need to convert the downloaded
+`feedback.json` to `feedback.js` by running the command:
+`./tnf generate catalog feedbackjs -f path-to-feedbackjson/feedback.json `
```
Comment (rdavid) [inline README.md:74]:
```suggestion
`./tnf generate catalog feedbackjs -f path-to-feedbackjson/feedback.json`
```

File: README.md (hunk 9 above)
Comment (greyerof) [reply README.md:68]: I agree.

File: README.md (hunk 10 above)
Comment (greyerof) [inline README.md:74]: Please update this example. This command is not under "generate catalog" as per your last changes. Also, you should include the new "-o" flag in the example

Comment (aabughosh) [reply README.md]: i did that

Comment (greyerof) [reply README.md]: Oh, my fault. Sorry.

File: README.md [hunk 11]
```diff
@@ -63,6 +63,18 @@ you we created a parser that is local. You can open it by copying the path /
 browser then upload the claim.json file of that run.
 ![overview](docs/assets/images/htmlpage.png)
 
+## Upload a previous feedback to the html parser page
+
+Users are able to add their feedback about their test run then download their
+own copy of their feedback via the parser HTML page.  Users can then re-use the
+feedback they have already entered in future runs.  To do this, users can
+upload their `feedback.json` file to the parser HTML page.  Alternatively,
```
Comment (greyerof) [inline README.md:71]: I don't understand this part: 

> Alternatively, users can place their `feedback.json` file in the path 
>  #your-code-path/cnf-certification-test/script/ and it needs to be named `feedback.js`."

 
Why do they need to place the file into that "script" folder? Users shouldn't care about the content of the cnf-`certification-test/results/html` folder, it's just for our code to use it in order to create the tar.gz file. This is my suggestion:

"In order to see that feedback in the HTML viewer/parser, it first needs to be converted to a javascript file called "feedback.js". Users can do this conversion using the `tnf` tool:
  ```
  ./tnf generate feedbackjs -f path/to/feedback.json -o my/output/folder
  ```
Where "my/output/folder" is a directory that must exist before running the command so the file "feedback.js" can be generated inside. Please note that this new javascript file must also be moved/copied to the same folder where the HTML files (results.html or results-embed.html) have been uncompressed so the web can use it. See the ["Results artifacts zip file" ](https://test-network-function.github.io/cnf-certification-test/test-output/#results-artifacts-zip-file) section of the "Test Output" tab in the documentation for more details."

**Review by greyerof — APPROVED** [review]

**Review by sebrandon1 — APPROVED** [review]

**Pull request merged**